data/random-teams.ts

```diff
@@ -1308,19 +1307,25 @@ export class RandomTeams {
 				!counter.technician ||
 				hasMove['tailslap'] ||
 				hasAbility['Punk Rock'] ||
+				// For Doubles Alolan Persian
 				movePool.includes('snarl')
 			);
 		case 'Tinted Lens':
 			return (
+				// For Sigilyph
 				hasMove['defog'] ||
-				(species.id === 'butterfree') ||
+				// For Butterfree
+				(hasMove['hurricane'] && hasAbility['Compound Eyes']) ||
```

Comment:
If this is "for Butterfree," why not hardcode it? Would you complain if Vivillon became legal and it got Compound Eyes with Hurricane? I personally think this is a good general cull, and would be fine with Compound Eyes Vivillon.

Reply:
vivillon doesn't have tinted lens. this was changed in order to reflect that in doubles and only in doubles, both butterfree and gmax butterfree want compound eyes. it still fulfills the purpose it had with a species call in singles.

Reply:
this would be fine as a general cull, though. i just noted what it was specifically for at this point in time

Reply:
ok